src/backend/InvenTree/generic/states/api.py

```diff
@@ -73,18 +74,8 @@ class AllStatusViews(StatusView):
 
     def get(self, request, *args, **kwargs):
         """Perform a GET request to learn information about status codes."""
-        data = {}
-
-        def discover_status_codes(parent_status_class, prefix=None):
-            """Recursively discover status classes."""
-            for status_class in parent_status_class.__subclasses__():
-                name = '__'.join([*(prefix or []), status_class.__name__])
-                data[name] = {
-                    'class': status_class.__name__,
-                    'values': status_class.dict(),
-                }
-                discover_status_codes(status_class, [name])
-
-        discover_status_codes(StatusCode)
```

Comment:
@wolflu05 do you have insight into why this was done this way? The git history shows that you authored it mainly. There was no coverage of the previous category and it seems everything works in PUI after the change.

Reply:
Yes, this was changed so that nested inherotors of the status code class are discovered too. This is used for machine status.

There is an abstract `MachineStatus` class available here, which every machine type has to inherit and define its custom set of status codes:
https://github.com/inventree/InvenTree/blob/master/src/backend/InvenTree/machine/machine_type.py#L25

Then all types are exposed unter `MachineStatus__xxx` where xxx is the name of the class inheriting the machine status class. The logic you removed is used to recursively detect status code inheritors and prepend the parent class name. You should see that the machine status codes were not shown properly now when navigating to the admin center > machines.

I actually thought I had added some tests for this, but it seems that I forgot. I would really appreciate if you could add one for me, as I personally won't have any time the next weeks to code something for inventree.

Reply:
I have amended the function and tests (API side) to cover this use case.